src/pages/tasks.page.tsx

```diff
@@ -0,0 +1,175 @@
+    const resizeObserver = new ResizeObserver(() =>{
+        const rectOuter = refOuter.current?.getBoundingClientRect();
+        const rectInner = refInner.current?.getBoundingClientRect();
+        // 外枠 < 内枠 なら省略表記
+        setIsEllipsis((rectOuter?.width ?? 0) < (rectInner?.width ?? 0));
+    });
+    const el = refOuter.current as Element;
+    resizeObserver.observe(el);
+    return () => resizeObserver.unobserve(el);
+    });
+
+    const {
+        children,
+        ...etc
+    } = props;
+    // 省略表記になるとき title を設定
+    const title = isEllipsis && { title: children };
+    return (
+        <OmitSpan ref={refOuter} {...etc}>
+            <span ref={refInner} {...title}>{children}</span>
+        </OmitSpan>
+    );
+}
+
+const TaskBottomButton = styled.div`
+    display: flex;
+    justify-content: space-between;
+    width: 210px;
+`
+const TaskBottomButtonChild  = styled.div`
+    background: #BF5745;
+    width: 3px;
+    height: 10px;
+    
+`
+const TextButton = styled.button`
+    margin-top: 20px;
+    color: #fff;
+    background: #BF5745;
+    width: 180px;
+    height: 50px;
+    text-align: center;
+`
+const DueButton = styled.button`
+    color: #fff;
+    background: #BF5745;
+    width: 210px;
+    height: 30px;
+    font-size: 10px;
+    text-align: right;
+`
+
+const DisplayGrid = styled.div`
+    display: grid;
+    grid-template-columns: 1fr 30px;
+`
+
+const EditBack = styled.div`
+    margin-top: 20px;
+    background: #BF5745;
+    height: 50px;
+`
+const DivMargin = styled.div`
+    margin-top: 10px;
+`
+function EditButton(){
+    return (
+        <EditBack>
+            <DivMargin>
+                <Link href="/">
+                    <IconContext.Provider value={{color: '#fff'}}>
+                        <BsFillPenFill />
+                    </IconContext.Provider>
+                </Link>
+            </DivMargin>
+        </EditBack>
+    )
+}
+function TaskButton(props: {taskTitle: string, dueYear: number, dueMonth: number, dueDay: number}){
+    return (
+        <>
+            <DisplayGrid>
+                <TextButton><Text>{props.taskTitle}</Text></TextButton>
+                <EditButton />
+            </DisplayGrid>
+            <div>
+                <DueButton>{props.dueYear}年{props.dueMonth}月{props.dueDay}日</DueButton>
+                <TaskBottomButton>
+                    <TaskBottomButtonChild />
+                    <TaskBottomButtonChild />
+                </TaskBottomButton>
+            </div>
+        </>
+    )
+}
+
+function TaskList(props: { taskList:  any[]}){
+    return (
+        <div>
+            {props.taskList.map(task => (<div key={task.taskID}><TaskButton taskTitle={task.taskName} dueYear={task.taskDueYear} dueMonth={task.taskDueMonth} dueDay={task.taskDueDay} /></div>))}
+        </div>
+    )
+}
+const Back = styled.div`
+    background-color: #1B2668;
+`
+const Center = styled.div`
+    display: flex;
+    justify-content: center;
+`
+export default function Home(){
```

Comment:
余談だけど ここの Home の部分はページのパスの名前にすることが多いかも
```suggestion
export default function Tasks(){
```